pkg/network/driver/nmstate/spec_test.go

```diff
@@ -0,0 +1,558 @@
+					MTU:        defaultMTU,
+					Ethtool:    defaultEthtool(),
+					LinuxStack: nmstate.LinuxIfaceStack{
+						IP4RouteLocalNet: pointer.P(false),
+						PortLearning:     pointer.P(false),
+					},
+					IPv4: nmstate.IP{Enabled: pointer.P(false)},
+					IPv6: nmstate.IP{Enabled: pointer.P(false)},
+				},
+			}))
+		})
+
+		ginkgo.It("deletes the IP addresses", func() {
+			err := nmState.Apply(&nmstate.Spec{Interfaces: []nmstate.Interface{
+				{
+					Name: dummyName,
+					IPv4: nmstate.IP{Enabled: pointer.P(false)},
+					IPv6: nmstate.IP{Enabled: pointer.P(false)},
+				},
+			}})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			status, err := nmState.Read()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(status.Interfaces).To(gomega.Equal([]nmstate.Interface{
+				{
+					Name:       dummyName,
+					Index:      1,
+					TypeName:   nmstate.TypeDummy,
+					State:      nmstate.IfaceStateUp,
+					MacAddress: macAddress0,
+					MTU:        defaultMTU,
+					Ethtool:    defaultEthtool(),
+					LinuxStack: nmstate.LinuxIfaceStack{
+						IP4RouteLocalNet: pointer.P(false),
+						PortLearning:     pointer.P(false),
+					},
+					IPv4: nmstate.IP{Enabled: pointer.P(false)},
+					IPv6: nmstate.IP{Enabled: pointer.P(false)},
+				},
+			}))
+		})
+
+		ginkgo.It("modifies the IP addresses", func() {
+			const (
+				ip4Addr1   = ip4Addr0 + "1"
+				ip4Prefix1 = ip4Prefix0 + 1
+				ip6Addr1   = ip6Addr0 + "1"
+				ip6Prefix1 = ip6Prefix0 + 1
+			)
+			err := nmState.Apply(&nmstate.Spec{Interfaces: []nmstate.Interface{
+				{
+					Name: dummyName,
+					IPv4: nmstate.IP{
+						Enabled: pointer.P(true),
+						Address: []nmstate.IPAddress{{
+							IP:        ip4Addr1,
+							PrefixLen: ip4Prefix1,
+						}},
+					},
+					IPv6: nmstate.IP{
+						Enabled: pointer.P(true),
+						Address: []nmstate.IPAddress{{
+							IP:        ip6Addr1,
+							PrefixLen: ip6Prefix1,
+						}},
+					},
+				},
+			}})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			status, err := nmState.Read()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(status.Interfaces).To(gomega.Equal([]nmstate.Interface{
+				{
+					Name:       dummyName,
+					Index:      1,
+					TypeName:   nmstate.TypeDummy,
+					State:      nmstate.IfaceStateUp,
+					MacAddress: macAddress0,
+					MTU:        defaultMTU,
+					Ethtool:    defaultEthtool(),
+					LinuxStack: nmstate.LinuxIfaceStack{
+						IP4RouteLocalNet: pointer.P(false),
+						PortLearning:     pointer.P(false),
+					},
+					IPv4: nmstate.IP{
+						Enabled: pointer.P(true),
+						Address: []nmstate.IPAddress{{
+							IP:        ip4Addr1,
+							PrefixLen: ip4Prefix1,
+						}},
+					},
+					IPv6: nmstate.IP{
+						Enabled: pointer.P(true),
+						Address: []nmstate.IPAddress{{
+							IP:        ip6Addr1,
+							PrefixLen: ip6Prefix1,
+						}},
+					},
+				},
+			}))
+		})
+
+		ginkgo.It("modifies its name (and removes the IP/s)", func() {
```

Comment:
Why do we care about IP's in the test? (and other test that doesnt touch IP's)
Is it because the interface will be set down and up?